x/sync/client.go

```diff
@@ -61,28 +61,27 @@ type client struct {
 }
 
 type ClientConfig struct {
-	NetworkClient       NetworkClient
+	NetworkClient       networkClient
```

Comment:
Since this is part of the interface, we should leave `NetworkClient` exported.

Reply:
Since what's part of what interface? You can create a new `networkClient` and put it in a `ClientConfig` even though `networkClient` is unexported

Reply:
I'm pretty sure this doesn't work since the methods in `networkClient` are not exported.

Reply:
What do you think doesn't work, specifically?